### PR TITLE
remove core-svc v9, fix core service shallow clone, fix working with datasets in the UI

### PR DESCRIPTION
**Pull request by Panaetius**
This is a bugfix release that includes the following:

- Remove support for core v9: those projects must be updated to v10. We kept v9 around for longer than the usual 6-months.
- Fix shallow clone in the core-service: listing datasets/workflows shouldn't time out as frequently anymore.
- Restore compatibility between renku-core and renku-ui to prevent errors on the datasets pages: a change in renku-core accidentally happened to break compatibility with the UI. The quickest and most reasonable fix was in the UI, hence a new bugfix release from UI too ([more details here](https://github.com/SwissDataScienceCenter/renku-ui/issues/2974))

/deploy extra-values=global.renku.cli_version=2.8.2rc2

### CODE REVIEW

**Comment by lorenzo-cavazzi** [comment]
@Panaetius do you mind if I disable the cypress tests for updating the `v(last - 1)` project?

**Comment by Panaetius** [comment]
@lorenzo-cavazzi no that's fine

**Review by lokijuhy — APPROVED** [review]

**Comment by jachro** [comment]
FYI, I'll also disable the v9 path on Project Update API but this will be in our release.

**Comment by lorenzo-cavazzi** [comment]
The `testDatasets` test requires a new UI version to pass; a bugfix release for the current version is coming "soon :tm: ".

**Review by aledegano — DISMISSED** [review]

File: CHANGELOG.rst [hunk 1]
```diff
@@ -2,18 +2,28 @@
 
 0.45.2
 ------
-This is a bugfix release fixing the process that removes expiring Project Access Tokens. 
+
+User-Facing Changes
+-------------------
+
+- **Core Service**: Removed support in for metadata v9 projects in the UI. Migration to v10 is now required.
+- **Core Service**: Fixed a bug where projects weren't cloned shallowly, leading go large projects not working properly on the platform.
```
Comment (rokroskar) [inline CHANGELOG.rst:10]:
```suggestion
- **Core Service**: Fixed a bug where projects weren't cloned shallowly, leading to large projects not working properly on the platform.
```

File: CHANGELOG.rst [hunk 2]
```diff
@@ -2,18 +2,28 @@
 
 0.45.2
 ------
-This is a bugfix release fixing the process that removes expiring Project Access Tokens. 
+
+User-Facing Changes
+-------------------
+
+- **Core Service**: Removed support in for metadata v9 projects in the UI. Migration to v10 is now required.
```
Comment (rokroskar) [inline CHANGELOG.rst:9]:
```suggestion
- **Core Service**: Removed support for metadata v9 projects in the UI. Migration to v10 is now required.
```

**Review by rokroskar — APPROVED** [review]

**Pull request merged**